.concourse/product-release/pipeline.template

```diff
@@ -0,0 +1,467 @@
+            export DOCKER_ORG=cap-staging
+            export QUIET_OUTPUT=true
+            make scf
+` }}
+
+{{- $test := `
+            # run test suites on cluster
+            # See: https://github.com/SUSE/catapult/wiki/Running-SCF-tests#kubecf
+            export KUBECF_TEST_SUITE="${TEST_SUITE:-smokes}"
+            export KUBECF_NAMESPACE="scf"
+            export SCF_TESTGROUP=true
+            export QUIET_OUTPUT=true
+            make tests-kubecf
+` }}
+
+{{- $stratos := `
+            # deploy stratos console & metrics
+            export QUIET_OUTPUT=true
+            make stratos metrics
+` }}
+
+{{- $upgrade := `
+            # upgrade kubecf
+            export SCF_CHART="$(readlink -f ../s3.kubecf-bundle/*.tgz)"
+            export SCF_OPERATOR=true
+            export QUIET_OUTPUT=true
+            make scf-chart # this takes the same chart as the deploy step for now
+            make scf-gen-config
+            make scf-upgrade
+` }}
+
+{{- $allbells := `
+        # enable as much options as possible
+        HA: true
+        EMBEDDED_UAA: true
+        AUTOSCALER: true
+` }}
+
+jobs:
+{{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
+
+{{ range $_, $OptionsFlag := (ds "OPTIONS").options -}} # for options
+
+{{ range $_, $EiriniFlag := (ds "EIRINI").eirini -}} # for eirini
+
+# {{$Backend}}-{{$OptionsFlag}}
+
+- name: deploy-{{$EiriniFlag}}-{{$Backend}}-{{$OptionsFlag}}
+  public: false
+  plan:
+  - get: s3.kubecf-bundle
+    trigger: true
+  - get: kubecf-master
+  - get: catapult
+  - task: deploy
+    privileged: true
+    timeout: 2h30m
+    input_mapping:
+      kubecf: kubecf-master
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: splatform/catapult
+      inputs:
+      - name: kubecf
+      - name: catapult
+      - name: s3.kubecf-bundle
+      outputs:
+      - name: output
+      params:
+        QUIET_OUTPUT: true
+        DEFAULT_STACK: cflinuxfs3
+        EKCP_HOST: ((ekcp-host))
+{{- if eq $EiriniFlag "eirini" }}
+        ENABLE_EIRINI: true
+{{ else }}
+        ENABLE_EIRINI: false
+{{- end }}
+{{- if eq $OptionsFlag "ha" }}
+        HA: true
+{{- end }}
+{{- if eq $OptionsFlag "all" }}
+{{- print $allbells }}
+{{- end }}
+        CLUSTER_NAME: {{$pipelineName}}-{{$EiriniFlag}}-{{$Backend}}-{{$OptionsFlag}}
+      run:
+        path: "/bin/bash"
+        args:
+          - -xce
+          - |
+            {{- print $obtain_ekcp }}
+            {{- if eq $Backend "caasp4" }}
+            {{- print $import_caasp4 }}
+            {{- end }}
+            {{- if eq $Backend "eks" }}
+            {{- print $import_eks }}
+            {{- end }}
+            {{- if eq $Backend "gke" }}
+            {{- print $import_gke }}
+            {{- end }}
+            {{- if eq $Backend "aks" }}
+            {{- print $import_aks }}
+            {{- end }}
+            {{- print $deploy }}
+
```

Comment:
You can apply my previous comments to the jobs below. Then it's going to be clearer to do a second revision pass on the specifics of each job.